lxd/storage/backend_lxd.go

```diff
@@ -2677,6 +2677,50 @@ func (b *lxdBackend) MountInstance(inst instance.Instance, op *operations.Operat
 	return mountInfo, nil
 }
 
+// DelegateInstance sets up an instance volume for delegation.
+func (b *lxdBackend) DelegateInstance(inst instance.Instance) error {
```

Comment:
I'm wondering about whether we should add a field to `MountInfo` of `PostHooks []func(inst instance.Instance) error` so that on `MountInstance()` (and we would need to add equivalent to `MountCustomVolume()` which I have been meaning to do for a while for consistency reasons anyway) a `MountInfo.PostHooks` field would be populated with a function that would be added to the device's returned post hooks (or for the root disk which is mounted in the lxc driver added to the post hooks list there instead) so that these hook functions would be run once the container is started.

This would avoid adding exporting these new functions in the backendLXD as well as avoiding the need to call them opportunistically when starting every disk device.

Instead the storage pool subsystem itself would decide whether delegation was needed as part of the Mount call and return an appropriate post-start hook to perform the delegation once the instance had started.

This feels like a way to achieve the same effect but without leaking the delegation concept (which is both instance type and storage pool driver specific) into the general interfaces. But I think the `DelegateVolume(vol Volume, pid int) error` at the storage driver level is fine as its sufficiently low level.

What do you think?

Reply:
That was my original plan but I was trying to keep this PR more readable :)

Anyway, doing that change now.

Reply:
Thanks!